uefi-mkconfig

```diff
@@ -236,24 +240,31 @@ main () {
 		done
 
 		# Sort kernel images
-		kernel_images="$(sort -uV <<< "$kernel_images_old") $(sort -uV <<< "$kernel_images_new")"
+		kernel_images="$(sort -urV <<< "$kernel_images_new") $(sort -urV <<< "$kernel_images_old")"
 
 		# Add path back to the kernel image
 		for kernel_image in $kernel_images; do
         		partition_efis_main+="$(find "$partition_mount" -name "$kernel_image") "
 		done
-		
+
 		# Create backup entry if it does not exist
 		partition_efis=$partition_efis_main
 		backup_entry
-		
+
 		# Add missiong efi entries for efi files that exist
 		partition_efis=$partition_efis_main
-		add_uefi_entries	
+		add_uefi_entries
 
 	done
 
-	einfo "Done"
+	#sorting boot item list into ascending boot order
+        einfo "Setting boot order..."
+	umc_items=$(efibootmgr -u|grep -Po '[Bb]oot0(1[0-9a-fA-F]{2}|200)\W{1,2}UMC'|grep -Po '[0-9a-fA-F]{4}'|sort |tr '\n' ','|sed 's/,$//g')
+	non_umc_items=$(efibootmgr -u|grep -Pv '[Bb]oot0(1[0-9a-fA-F]{2}|200)\W{1,2}UMC'|grep -Po '[Bb]oot[0-9a-fA-F]{4}'|grep -Po '[0-9a-fA-F]{4}'|sort |tr '\n' ','|sed 's/,$//g')
+	itemlist="$umc_items,$non_umc_items"
+	efibootmgr -q -o "$itemlist"
+
+        einfo "Done"
```

Comment:
If we are explicitly setting the order afterwards anyway, then I think it makes sense to use ` --create-only` for `efibootmgr` instead of ` --create`. This saves a write to an efivar, and also removes the need for the `if` on line 113 to 119.